### PR TITLE
flakey videoAPI test (#4947)

### PR DESCRIPTION
**Pull request by bjuncek**
Finding the cause of the flaky test as cannot repro locally with either of current settings:
python 3.7/8/9
ffmpeg 4.2.2

- [x] repro on ci?
_(everything seems to work - I'll re trigger test few more times just to be sure)_
- [x] remove IO bottleneck by creating a resource pool
- [x] profit?

### CODE REVIEW

File: test/test_videoapi.py [hunk 1]
```diff
@@ -57,34 +57,66 @@ def test_frame_reading(self):
...
+                    av_frames.append(torch.tensor(av_frame.to_rgb().to_ndarray()).permute(2, 0, 1))
+                    av_pts.append(av_frame.pts * av_frame.time_base)
+                av_reader.close()
 
-                    av_array = torch.tensor(av_frame.to_rgb().to_ndarray()).permute(2, 0, 1)
-                    vr_array = vr_frame["data"]
-                    mean_delta = torch.mean(torch.abs(av_array.float() - vr_array.float()))
+                # get vr frames
+                video_reader = VideoReader(full_path, "video")
+                for vr_frame in video_reader:
```
Comment (prabhat00155) [inline test/test_videoapi.py:73]: Is there any advantage of have two separate for loops? We could just use one like here: https://github.com/pytorch/vision/blob/e3767f8aa51921460c89e1d396ccce097cebc9d5/test/test_video_gpu_decoder.py#L33 and do something like line 84 on a frame by frame basis and not have to store a list of frames.

File: test/test_videoapi.py [hunk 2]
```diff
@@ -57,34 +57,66 @@ def test_frame_reading(self):
...
+                # get av frames
+                for av_frame in av_reader.decode(av_reader.streams.audio[0]):
+                    av_frames.append(torch.tensor(av_frame.to_ndarray()).permute(1, 0))
+                    av_pts.append(av_frame.pts * av_frame.time_base)
+                av_reader.close()
 
-                    max_delta = torch.max(torch.abs(av_array.float() - vr_array.float()))
+                # get vr frames
+                video_reader = VideoReader(full_path, "audio")
+                for vr_frame in video_reader:
```
Comment (prabhat00155) [inline test/test_videoapi.py:108]: Same as above.

File: test/test_videoapi.py [hunk 3]
```diff
@@ -57,34 +57,66 @@ def test_frame_reading(self):
...
+                av_frames, vr_frames = [], []
+                av_pts, vr_pts = [], []
+                # get av frames
                 for av_frame in av_reader.decode(av_reader.streams.video[0]):
-                    vr_frame = next(video_reader)
-
-                    assert float(av_frame.pts * av_frame.time_base) == approx(vr_frame["pts"], abs=0.1)
+                    av_frames.append(torch.tensor(av_frame.to_rgb().to_ndarray()).permute(2, 0, 1))
+                    av_pts.append(av_frame.pts * av_frame.time_base)
+                av_reader.close()
```
Comment (prabhat00155) [inline test/test_videoapi.py:69]: Minor comment: If you use `with`, you won't need to worry about this.

**Review by prabhat00155 — COMMENTED** [review]
Thanks @bjuncek! I have a couple of comments. Let me know what you think.

File: test/test_videoapi.py (hunk 3 above)
Comment (bjuncek) [reply test/test_videoapi.py:69]: fair, addressed

File: test/test_videoapi.py (hunk 1 above)
Comment (bjuncek) [reply test/test_videoapi.py:73]: Theoretically, having separate loops, and closing the file pointers between the readers should increase the stability of the tests as there is no instance in which there are multiple pointers to the same file.\TBH figured it's one of those things that can't hurt, but might increase stability

Comment (prabhat00155) [reply test/test_videoapi.py]: Do you suspect that being the reason for the tests being flaky?

Comment (bjuncek) [reply test/test_videoapi.py]: Not 100% sure, but I was told it was a good practice, especially when dealing with C++ memcpys.
